idea-plugin/src/main/kotlin/arrow/meta/ide/testing/IdeTest.kt

```diff
@@ -0,0 +1,38 @@
+package arrow.meta.ide.testing
+
+import arrow.meta.ide.testing.dsl.IdeTestSyntax
+
+data class IdeTest(
+  val code: Source,
+  val assert: Companion.() -> Assert
+) {
+  companion object : Assert.Syntax, IdeTestSyntax
+}
+
+sealed class Assert {
+  sealed class IdeResolution<A> : Assert() {
+    data class Resolves<A>(val f: (A) -> Boolean) : IdeResolution<A>()
+    data class FailsWith<A>(val f: (A) -> Boolean) : IdeResolution<A>()
+    object Fails : IdeResolution<Nothing>()
+  }
+
+  object Empty : Assert()
+
+  interface Syntax {
+    val empty: Assert
+      get() = Empty
+    val fails: Assert
+      get() = IdeResolution.Fails
+    val resolves: Assert
+      get() = IdeResolution.Resolves<Int> { true }
+
+    fun <A> failsWith(f: (A) -> Boolean): Assert = IdeResolution.FailsWith(f)
+    fun <A> resolves(f: (A) -> Boolean): Assert = IdeResolution.Resolves(f)
+  }
+}
+
+
+// copied from testing-plugin couldn't import the project
```

Comment:
/cc @rachelcarmena

Reply:
Thanks for the mention @raulraja !

I going to download the branch to check the issue, because `testing-plugin` is already used in `compiler-plugin`.

Reply:
Got it! :tada: 

I've seen this line in `gradle.properties`:

```
// testCompile project(":testing-plugin")
```

When using `testCompile`, that dependency won't be available for source code located in `src/main` but `src/test`.

This piece of source code is under `src/main`.

**NOTE**

`testing-plugin` is though to write tests so it makes sense to find something like:

```
testImplementation project(":testing-plugin")
```

because it's needed to compile tests and also needed in test runtime.